tools/rostopic/src/rostopic/__init__.py

```diff
@@ -300,28 +300,37 @@ def msgevalgen(pattern):
 
     # use slice arguments if present
     array_index_or_slice_object = None
-    index = msg_attribute.find('[')
-    if index != -1:
-        if not msg_attribute.endswith(']'):
+    rec_msgeval = None
+    index_start = msg_attribute.find('[')
+    if index_start != -1:
+        index_end = msg_attribute.find(']')
```

Comment:
Shouldn't the find search start from `index_start`? Or at least check the order of the start and end indices afterwards?